conversation_reconstruction/dataflow/write_page_states_to_spanner/dataflow_main.py

```diff
@@ -56,31 +62,43 @@ def run(known_args, pipeline_args):
 
   with beam.Pipeline(options=pipeline_options) as p:
     # Main Pipeline
-    p = (p | beam.io.ReadFromText(known_args.input_storage)
-         | beam.ParDo(WriteToSpanner(), known_args.spanner_instance, known_args.spanner_database,
-                      known_args.spanner_table, known_args.spanner_table_columns))
+    if known_args.input_storage is not None:
+      # Read from cloud storage.
+      input_data = (p | beam.io.ReadFromText(known_args.input_storage))
+    else:
+      # Read from BigQuery.
+      input_data = (p | beam.io.Read(beam.io.BigQuerySource(
+          query="SELECT * FROM %s" % known_args.bigquery_table, use_standard_sql=True)))
+    p = (input_data | beam.ParDo(WriteToSpanner(known_args.spanner_instance, known_args.spanner_database,
+                                     known_args.spanner_table, known_args.spanner_table_columns), known_args.input_storage))
 
 class WriteToSpanner(beam.DoFn):
```

Comment:
Add a docstring, and specify assumptions. e.g. that the table doesn't already exist.